src/xmppjs/XJSInstance.ts

```diff
@@ -117,11 +120,16 @@ export class XmppJsInstance extends EventEmitter implements IBifrostInstance {
     }
 
     public xmppWriteToStream(xmlMsg: any) {
```

Comment:
I worry that this might hide errors more than fix them – I'd rather give this a stronger type signature and make sure we know what we're dealing with.

Looking through the code though, this seems to always be called either with `Element` (across `ServiceHandler`) or once with a `StzaPresenceItem` (in `XJSGateway.remoteLeft`), at least on current master. Looking at the latter, I'm not sure if `toString()` even turns it into a proper XML string (`xml()` seems to be responsible for that).

How about we make the type checks stronger here, and make sure we're handling things correctly?

Reply:
No problem by my side, my change was just to fix broken promises as this.xmpp.write() only accepts either strings or a buffer after the dependencies bump (previously it worked just fine with elements).